alacritty/src/renderer/mod.rs

```diff
@@ -218,3 +220,48 @@ impl Renderer {
         }
     }
 }
+
+struct GlExtensions;
+
+impl GlExtensions {
+    /// Check if the given `extension` is supported.
+    ///
+    /// This function will lazyly load OpenGL extensions.
+    fn contains(extension: &str) -> bool {
+        static mut OPENGL_EXTENSIONS: Option<HashSet<&'static str>> = None;
+        static EXTENSIONS_LOADED: Once = Once::new();
```

Comment:
Global static state isn't super common in Rust code usually, but honestly for this kinda stuff I think it makes things pretty convenient. Especially because the underlying OpenGL structures are all static anyway.

Reply:
Yeah, but we want a singleton here, since we expose global static immutable object.